cylc/flow/scheduler.py

```diff
@@ -1896,3 +1900,27 @@ async def handle_exception(self, exc: Exception) -> NoReturn:
         """
         await self.shutdown(exc)
         raise exc from None
+
+    def validate_finalcp(self):
+        """Warn if Stop Cycle point is on or after the final cycle point
+        """
+        if not self.options.fcp:
+            return
+        if (
+            (
+                self.config.cycling_type == INTEGER_CYCLING_TYPE
+                and self.options.stopcp > (self.options.fcp)
+            )
+            or
+            (
+                self.config.cycling_type == ISO8601_CYCLING_TYPE
+                and (
+                    ISO8601Point(self.options.stopcp) >
+                    ISO8601Point(self.options.fcp)
+                )
+            )
+        ):
```

Comment:
No need to handle the cycling modes differently, `get_point()` takes care of that.
```suggestion
        if get_point(self.options.stopcp) > get_point(self.config.final_point):
```

Reply:
This is lovely. Thank you @MetRonnie